### DISCUSSION TITLE
oqtane with syncfusion

### DISCUSSION THREAD

**Discussion by pnicosia80** [oqtane/oqtane.framework#3066] [Q&A]
Good morning,

I tried to setup my oqtane project with syncfusion as given at below link:

https://www.syncfusion.com/succinctly-free-ebooks/custom-blazor-oqtane-modules-succinctly/add-syncfusion

Unfortunately the components don't work for me and I see them like this:

![erroreSyncfusion](https://github.com/oqtane/oqtane.framework/assets/10197467/752b9afd-1885-41cb-986f-174eff885251)

I used the nuget packages version 22.1.39, the Syncfusion_Blazor_Template_Studio and also the syncfusionessentialstudiowebinstaller.
Oqtane is updated to 4.0.1.

Can you help me please?

Greetings,

Pietro

**Comment by leigh-pointer** [comment]
@adrianus80 I think a good place to start would be to read this blog post https://blazorhelpwebsite.com/ViewBlogPost/47

**Comment by W6HBR** [comment]
You're either missing the references to the bootstrap & material CSS files, or you're missing the files themselves.

You need to copy the static web assets from the NuGet package using the instructions from the link @leigh-pointer posted above. You then need to include the references to the stylesheet in your razor file.

In your Index.razor, add the two references to the Syncfusion css files:
```
    public override List<Resource> Resources => new List<Resource>()
    {
        new Resource { ResourceType = ResourceType.Stylesheet, Url = ModulePath() + "Module.css" },
        new Resource { ResourceType = ResourceType.Script, Url = ModulePath() + "Module.js" },
        new Resource { ResourceType = ResourceType.Stylesheet, Url = "_content/Syncfusion.Blazor/styles/bootstrap4.css" },
        new Resource { ResourceType = ResourceType.Stylesheet, Url = "_content/Syncfusion.Blazor/styles/material.css" }
    };
```

**Comment by markagama** [comment]
Hi @adrianus80, did you get the example fully working with Oqtane 4? I'm starting with the same example, and would be happy to collaborate.

After making this fix, I get an error about an unknown 'source' - any ideas?

**Reply by W6HBR** [reply]
@markagama Can you give some additional details about this error? Maybe a copy of the exact error?
I've done a few projects with Syncfusion controls in Oqtane 4 now and it is a little different (but not too much) from the original instructions.  Maybe with some more specifics about your error, I can help.

**Reply by markagama** [reply]
thx for responding @W6HBR 

I'm trying to get Michael's example working - seems to be a good place to start.
I've fixed the resource issue but now have a database issue that I think is related to the shift from Oqtane 2 to 4
I get an error related to helpdeskcontroller.cs line 40. The code at this line is

            var HelpDeskTickets = _HelpdeskRepository.GetSyncfusionModuleHelpDeskTickets(int.Parse(entityid))
                .Where(x => x.CreatedBy == username)
                .OrderBy(x => x.HelpDeskTicketId)
                .ToList();

The error is:
Microsoft.EntityFrameworkCore.Database.Command: Information: Executed DbCommand (1ms) [Parameters=[@__entityName_0='?' (Size = 4000)], CommandType='Text', CommandTimeout='30']
SELECT [s].[SettingId], [s].[CreatedBy], [s].[CreatedOn], [s].[EntityId], [s].[EntityName], [s].[IsPrivate], [s].[ModifiedBy], [s].[ModifiedOn], [s].[SettingName], [s].[SettingValue]
FROM [Setting] AS [s]
WHERE [s].[EntityName] = @__entityName_0
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\share\Repos\oqtane.framework\Oqtane.Server\bin\Debug\net7.0\Microsoft.IdentityModel.Protocols.OpenIdConnect.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\share\Repos\oqtane.framework\Oqtane.Server\bin\Debug\net7.0\Microsoft.IdentityModel.Protocols.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\share\Repos\oqtane.framework\Oqtane.Server\bin\Debug\net7.0\Microsoft.IdentityModel.Logging.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.10\System.Net.Security.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executing endpoint '/_Host'
Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker: Information: Route matched with {page = "/_Host", action = "", controller = ""}. Executing page /_Host
Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker: Information: Executing handler method Oqtane.Pages.HostModel.OnGet - ModelState is Valid
Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker: Information: Executed handler method OnGet, returned result Microsoft.AspNetCore.Mvc.RazorPages.PageResult.
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\7.0.10\Microsoft.AspNetCore.DataProtection.Extensions.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.10\Microsoft.CSharp.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.10\System.Net.Http.Json.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.10\System.Net.Sockets.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.10\System.Net.NameResolution.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request starting HTTP/1.1 GET http://localhost:44357/api/Installation/installed/?path= - -
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executing endpoint 'Oqtane.Controllers.InstallationController.IsInstalled (Oqtane.Server)'
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Route matched with {action = "IsInstalled", controller = "Installation", page = ""}. Executing controller action with signature Oqtane.Models.Installation IsInstalled(System.String) on controller Oqtane.Controllers.InstallationController (Oqtane.Server).
Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor: Information: Executing ObjectResult, writing value of type 'Oqtane.Models.Installation'.
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Executed action Oqtane.Controllers.InstallationController.IsInstalled (Oqtane.Server) in 137.8444ms
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executed endpoint 'Oqtane.Controllers.InstallationController.IsInstalled (Oqtane.Server)'
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request finished HTTP/1.1 GET http://localhost:44357/api/Installation/installed/?path= - - - 200 - application/json;+charset=utf-8 280.3773ms
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request starting HTTP/1.1 GET http://localhost:44357/api/User/name/Mark?siteid=1 - -
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executing endpoint 'Oqtane.Controllers.UserController.Get (Oqtane.Server)'
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Route matched with {action = "Get", controller = "User", page = ""}. Executing controller action with signature Oqtane.Models.User Get(System.String, System.String) on controller Oqtane.Controllers.UserController (Oqtane.Server).
Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor: Information: Executing ObjectResult, writing value of type 'Oqtane.Models.User'.
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Executed action Oqtane.Controllers.UserController.Get (Oqtane.Server) in 541.5277ms
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executed endpoint 'Oqtane.Controllers.UserController.Get (Oqtane.Server)'
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request finished HTTP/1.1 GET http://localhost:44357/api/User/name/Mark?siteid=1 - - - 200 - application/json;+charset=utf-8 675.9458ms
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request starting HTTP/1.1 GET http://localhost:44357/api/Sync/00010101000000000 - -
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executing endpoint 'Oqtane.Controllers.SyncController.Get (Oqtane.Server)'
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Route matched with {action = "Get", controller = "Sync", page = ""}. Executing controller action with signature Oqtane.Models.Sync Get(System.String) on controller Oqtane.Controllers.SyncController (Oqtane.Server).
Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor: Information: Executing ObjectResult, writing value of type 'Oqtane.Models.Sync'.
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Executed action Oqtane.Controllers.SyncController.Get (Oqtane.Server) in 21.671ms
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executed endpoint 'Oqtane.Controllers.SyncController.Get (Oqtane.Server)'
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request finished HTTP/1.1 GET http://localhost:44357/api/Sync/00010101000000000 - - - 200 - application/json;+charset=utf-8 89.2654ms
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request starting HTTP/1.1 GET http://localhost:44357/api/Site/1 - -
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executing endpoint 'Oqtane.Controllers.SiteController.Get (Oqtane.Server)'
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Route matched with {action = "Get", controller = "Site", page = ""}. Executing controller action with signature Oqtane.Models.Site Get(Int32) on controller Oqtane.Controllers.SiteController (Oqtane.Server).
Microsoft.AspNetCore.Mvc.Infrastructure.ObjectResultExecutor: Information: Executing ObjectResult, writing value of type 'Oqtane.Models.Site'.
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Executed action Oqtane.Controllers.SiteController.Get (Oqtane.Server) in 357.625ms
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executed endpoint 'Oqtane.Controllers.SiteController.Get (Oqtane.Server)'
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request finished HTTP/1.1 GET http://localhost:44357/api/Site/1 - - - 200 - application/json;+charset=utf-8 429.1551ms
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request starting HTTP/1.1 GET http://localhost:44357/api/File/1 - -
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executing endpoint 'Oqtane.Controllers.FileController.Get (Oqtane.Server)'
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Route matched with {action = "Get", controller = "File", page = ""}. Executing controller action with signature Oqtane.Models.File Get(Int32) on controller Oqtane.Controllers.FileController (Oqtane.Server).
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request starting HTTP/1.1 GET http://localhost:44357/api/Helpdesk?username=Mark&authmoduleid=30 - -
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executing endpoint 'Syncfusion.Module.Helpdesk.Controllers.HelpdeskController.Get (Syncfusion.Module.Helpdesk.Server.Oqtane)'
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Route matched with {action = "Get", controller = "Helpdesk", page = ""}. Executing controller action with signature System.Collections.Generic.IEnumerable`1[Syncfusion.Module.Helpdesk.Models.SyncfusionModuleHelpDeskTickets] Get(System.String, System.String) on controller Syncfusion.Module.Helpdesk.Controllers.HelpdeskController (Syncfusion.Module.Helpdesk.Server.Oqtane).
Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker: Information: Executed action Syncfusion.Module.Helpdesk.Controllers.HelpdeskController.Get (Syncfusion.Module.Helpdesk.Server.Oqtane) in 28.1577ms
Microsoft.AspNetCore.Routing.EndpointMiddleware: Information: Executed endpoint 'Syncfusion.Module.Helpdesk.Controllers.HelpdeskController.Get (Syncfusion.Module.Helpdesk.Server.Oqtane)'
Exception thrown: 'System.ArgumentNullException' in System.Private.CoreLib.dll
Exception thrown: 'System.ArgumentNullException' in System.Private.CoreLib.dll
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware: Error: An unhandled exception has occurred while executing the request.

System.ArgumentNullException: Value cannot be null. (Parameter 's')
   at System.Int32.Parse(String s)
   at Syncfusion.Module.Helpdesk.Controllers.HelpdeskController.Get(String username, String entityid) in C:\share\Repos\Syncfusion.Module.Helpdesk\Server\Controllers\HelpdeskController.cs:line 40
   at lambda_method1440(Closure, Object, Object[])
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Oqtane.Infrastructure.JwtMiddleware.Invoke(HttpContext context) in C:\share\Repos\oqtane.framework\Oqtane.Server\Infrastructure\Middleware\JwtMiddleware.cs:line 90
   at Oqtane.Infrastructure.TenantMiddleware.Invoke(HttpContext context) in C:\share\Repos\oqtane.framework\Oqtane.Server\Infrastructure\Middleware\TenantMiddleware.cs:line 67
   at Microsoft.AspNetCore.Localization.RequestLocalizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)

**Comment by W6HBR** [comment]
I would suggest doing a simpler module since the HelpDesk is a bit more complex and the instructions are no longer correct (but they are very close).  I didn't really have time to write up a complete how-to, but I've made a simple module with Syncfusion added that I can share.

I started by creating a new module in Oqtane and saved a copy of that source, then added Syncfusion to it and also saved that source.  With the two projects zipped up, you can easily compare what changed to get Syncfusion Blazor working.  I'd suggest unzipping both and using something like WinMerge to compare the code side by side.

You can find the module project files here:  https://github.com/W6HBR/Simple-Syncfusion-Demo

![image](https://github.com/oqtane/oqtane.framework/assets/7365166/fb0c7420-f486-4c71-8dad-2abd7f2bcf99)

**Reply by markagama** [reply]
thank you @W6HBR this is exactly what I needed to get moving. Also @sbwalker has provided a pointer at the two issues: the resource fix and the fix for an empty table.

I appreciate the assistance and will be moving ahead now with several modules for my project. Hopefully, I can assist others as I learn more.

**Reply by ADefWebserver** [reply]
@W6HBR Thanks for doing this!

**Reply by W6HBR** [reply]
Glad to help. 
@ADefWebserver Michael, your examples were exactly what I needed when I was getting started using the controls, so thank you for that!

**Reply by ADefWebserver** [reply]
@W6HBR I put a link to your answer on the Syncfusion repo.

**Reply by oqtanecms** [reply]
> thank you @W6HBR this is exactly what I needed to get moving. Also @sbwalker has provided a pointer at the two issues: the resource fix and the fix for an empty table.
> 
> I appreciate the assistance and will be moving ahead now with several modules for my project. Hopefully, I can assist others as I learn more.
Are you using SfToast in your module? It seems that SfToast content cannot be shown in oqtane module.I tested SfToast with oqtane 5.21,Syncfusion.Blazor is 26.1.35.
any help you provide is greatly appreciated.
<div class="col-lg-12 control-section toast-default-section" style="height:100px;">
    <SfToast ID="toast_default" @ref="ToastObj" Title="Adaptive Tiles Meeting" Content="@ToastContent" Timeout="5000" Icon="e-meeting">
        <ToastPosition X="@ToastPosition"></ToastPosition>
    </SfToast>
    <div class="col-lg-12 col-sm-12 col-md-12 center">
        <div id="toastBtnDefault" style="margin: auto;text-align: center">
            <button class="e-btn" @onclick="@ShowOnClick">Show Toasts2</button>
            <button class="e-btn" @onclick="@HideOnClick">Hide All</button>
        </div>
    </div>
</div>

**Reply by W6HBR** [reply]
@oqtanecms 

I just added this to a project that is in 5.2.1 and had no issue.

![image](https://github.com/user-attachments/assets/dd5fe627-29ec-4206-9d5f-6131b40e4bd1)


Please provide some additional details about how you are implementing this and maybe I can see where the issue is.

**Reply by oqtanecms** [reply]
Thanks for your reply. Now it can be displayed, but sometimes it can't be displayed. I don't know why, but the display position seems to be inappropriate. Is there any way to make it appear above the navigation menu?
![SfToast1](https://github.com/user-attachments/assets/377badb0-bc7f-4dc5-8843-794d1e1a787b)
![SfToast2](https://github.com/user-attachments/assets/320583b6-7937-4376-a109-2fdf1de6ba3c)

**Reply by oqtanecms** [reply]
When it cannot be displayed, you can capture the following error message：
![SfToast3](https://github.com/user-attachments/assets/1d49458e-296a-4839-816d-46ddc4a1c7b6)

**Reply by oqtanecms** [reply]
Add a style definition, it seems to work fine the first time, but it will not display correctly when clicked again, which feels very strange.
![SfToast5](https://github.com/user-attachments/assets/7b66c63d-e0f9-413f-891b-821123d8cac2)
![SfToast4](https://github.com/user-attachments/assets/6b990baf-001c-461e-8c79-959399f779bc)
![SfToast6](https://github.com/user-attachments/assets/422fa49e-6d48-4575-acc1-9aab08299fd9)

**Comment by leigh-pointer** [comment]
It might be worth looking at the syncfusion oqtane templates
________________________________

**Comment by ADefWebserver** [comment]
I just updated: https://github.com/ADefWebserver/Oqtane.SyncfusionHelpDesk to work with the latest Oqtane and .Net 8
